int32 is_ground
int32 is_del
int32 is_infect
